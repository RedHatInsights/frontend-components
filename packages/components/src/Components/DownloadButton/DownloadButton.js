import React, { Component } from 'react';
import { Dropdown, DropdownToggle, DropdownItem } from '@patternfly/react-core';
import { ExportIcon, IconSize } from '@patternfly/react-icons';
import PropTypes from 'prop-types';

class DownloadButton extends Component {
    state = {
        isOpen: false
    }

    onToggle = (isOpen) => {
        this.setState({
            isOpen
        });
    }

    onSelect = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    render() {
        const { isOpen } = this.state;
        const { extraItems, onSelect, isDisabled, ...props } = this.props;
        return (
            <Dropdown
                { ...props }
                isPlain
                onSelect={ this.onSelect }
                toggle={
                    <DropdownToggle
                        iconComponent={null}
                        onToggle={ this.onToggle }
                    >
                        <ExportIcon size={ IconSize.sm } />
                    </DropdownToggle>
                }
                isOpen={ isOpen }
                dropdownItems={ [
                    <DropdownItem key="download-csv" component="button" onClick={ event => onSelect(event, 'csv') } isDisabled={isDisabled}>
                        Export to CSV</DropdownItem>,
                    <DropdownItem key="download-json" component="button" onClick={ event => onSelect(event, 'json') } isDisabled={isDisabled}>
                        Export to JSON</DropdownItem>,
                    ...extraItems
                ] }
            />
        );
    }
}

DownloadButton.propTypes = {
    extraItems: PropTypes.arrayOf(PropTypes.node),
    onSelect: PropTypes.func
};
DownloadButton.defaultProps = {
    extraItems: [],
    onSelect: () => undefined
};

export default DownloadButton;
