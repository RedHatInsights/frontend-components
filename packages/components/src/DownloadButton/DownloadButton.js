import { Dropdown, DropdownItem, DropdownToggle, Tooltip } from '@patternfly/react-core';
import React, { Component } from 'react';

import { ExportIcon } from '@patternfly/react-icons';
import PropTypes from 'prop-types';

class DownloadButton extends Component {
  state = {
    isOpen: false,
  };

  onToggle = (isOpen) => {
    this.setState({
      isOpen,
    });
  };

  onSelect = () => {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  };

  conditionallyTooltip = (children) => {
    const { tooltipText } = this.props;
    return <React.Fragment>{tooltipText ? <Tooltip content={tooltipText}>{children}</Tooltip> : children}</React.Fragment>;
  };

  render() {
    const { isOpen } = this.state;
    const { extraItems, onSelect, isDisabled, tooltipText, ...props } = this.props;
    return (
      <React.Fragment>
        {this.conditionallyTooltip(
          <Dropdown
            {...props}
            isPlain
            onSelect={this.onSelect}
            toggle={
              <DropdownToggle toggleIndicator={null} onToggle={this.onToggle} isDisabled={isDisabled} ouiaId="Export">
                <ExportIcon size="sm" />
              </DropdownToggle>
            }
            isOpen={isOpen}
            ouiaId="Export"
            dropdownItems={[
              <DropdownItem
                key="download-csv"
                ouiaId="DownloadCSV"
                component="button"
                onClick={(event) => onSelect(event, 'csv')}
                isDisabled={isDisabled}
              >
                Export to CSV
              </DropdownItem>,
              <DropdownItem
                key="download-json"
                ouiaId="DownloadJSON"
                component="button"
                onClick={(event) => onSelect(event, 'json')}
                isDisabled={isDisabled}
              >
                Export to JSON
              </DropdownItem>,
              ...extraItems,
            ]}
          />
        )}
      </React.Fragment>
    );
  }
}

DownloadButton.propTypes = {
  extraItems: PropTypes.arrayOf(PropTypes.node),
  tooltipText: PropTypes.node,
  onSelect: PropTypes.func,
  isDisabled: PropTypes.bool,
};
DownloadButton.defaultProps = {
  extraItems: [],
  onSelect: () => undefined,
};

export default DownloadButton;
