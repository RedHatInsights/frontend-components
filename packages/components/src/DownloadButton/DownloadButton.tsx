import { Dropdown, DropdownItem, DropdownProps, DropdownToggle, Tooltip } from '@patternfly/react-core';
import React, { useState } from 'react';

import { ExportIcon } from '@patternfly/react-icons';

export interface DownloadButtonProps extends Omit<DropdownProps, 'onSelect' | 'toggle' | 'dropdownItems'> {
  /**
   * Additional JSX elements rendered as dropdown options
   */
  extraItems?: React.ReactElement[];
  /**
   * Text to appear in the tooltip
   */
  tooltipText?: React.ReactNode;
  /**
   * Action the button will take when selected
   */
  onSelect?: (event: MouseEvent | React.MouseEvent<any, MouseEvent> | React.KeyboardEvent<Element>, format: 'csv' | 'json') => void;
  /**
   * Determines if the button is disabled or not
   */
  isDisabled?: boolean;
}

/**
 * Download Button is a button component specifically made to be used for any download logic
 */

const DownloadButton: React.FunctionComponent<DownloadButtonProps> = ({
  extraItems = [],
  onSelect = () => undefined,
  isDisabled,
  tooltipText,
  ...props
}) => {
  const [isOpen, setIsOpen] = useState(false);

  const onToggle = (isOpen: boolean) => setIsOpen(isOpen);

  const internalOnSelect = () => setIsOpen((prev) => !prev);

  const conditionallyTooltip = (children: React.ReactElement) => {
    return <React.Fragment>{tooltipText ? <Tooltip content={tooltipText}>{children}</Tooltip> : children}</React.Fragment>;
  };

  return (
    <React.Fragment>
      {conditionallyTooltip(
        <Dropdown
          isPlain
          {...props}
          onSelect={internalOnSelect}
          toggle={
            <DropdownToggle aria-label="Export" toggleIndicator={null} onToggle={onToggle} isDisabled={isDisabled} ouiaId="Export">
              <ExportIcon size="sm" />
            </DropdownToggle>
          }
          isOpen={isOpen}
          ouiaId="Export"
          dropdownItems={[
            <DropdownItem
              key="download-csv"
              ouiaId="DownloadCSV"
              component="button"
              onClick={(event) => onSelect(event, 'csv')}
              isDisabled={isDisabled}
            >
              Export to CSV
            </DropdownItem>,
            <DropdownItem
              key="download-json"
              ouiaId="DownloadJSON"
              component="button"
              onClick={(event) => onSelect(event, 'json')}
              isDisabled={isDisabled}
            >
              Export to JSON
            </DropdownItem>,
            ...extraItems,
          ]}
        />
      )}
    </React.Fragment>
  );
};

export default DownloadButton;
