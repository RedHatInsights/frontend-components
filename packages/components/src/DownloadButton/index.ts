export { default } from './DownloadButton';
export { default as DownloadButton } from './DownloadButton';
export * from './DownloadButton';
