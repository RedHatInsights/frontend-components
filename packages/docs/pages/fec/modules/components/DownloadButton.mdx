import ExampleComponent from '@docs/example-component'

# DownloadButton
Download Button is a button component specifically made to be used for any download logic
<ExampleComponent source="DownloadButton/downloadbutton-example.js" name="downloadbutton example.js" />

## Props

|name|type|default|description|
|----|----|-------|-----------|
|extraItems|React.ReactElement[]|`[]`|Additional JSX elements rendered as dropdown options|
|tooltipText|React.ReactNode||Text to appear in the tooltip|
|onSelect|(event:MouseEvent&#124;React.MouseEvent&lt;any,MouseEvent&gt;&#124;React.KeyboardEvent&lt;Element&gt;,format:'csv'&#124;'json')=&gt;void|`() => undefined`|Action the button will take when selected|
|isDisabled|<code>&#123;<br/><span className="default-prop-spacer"></span><span className="default-prop-spacer"></span><span className="default-prop-spacer"></span> "name": "boolean"<br />&#125;</code>||Determines if the button is disabled or not|



