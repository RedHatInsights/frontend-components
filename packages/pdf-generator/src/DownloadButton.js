import React from 'react';
import { PDFDownloadLink } from '@react-pdf/renderer';
import PropTypes from 'prop-types';
import {
    Page,
    Text,
    View,
    Document,
    PDFViewer
} from '@react-pdf/renderer';
import RHLogo from './components/Logo';
import { DateFormat } from '../../components/src/Components/DateFormat';
import { customTitle } from './utils/text';
import styles from './utils/styles';

class DownloadButton extends React.Component {
    render() {
        const {
            pages,
            fileName,
            label,
            type,
            style,
            title,
            isPreview,
            reportName,
            ...props
        } = this.props;
        const appliedStyles = styles(style);

        const document = <Document style={{
            height: '100%'
        }}>
            {pages.map((page, key) => (
                <Page key={key} style={[ appliedStyles.page, appliedStyles.text ]}>
                    {<React.Fragment>
                        <View style={appliedStyles.headerContainer}>
                            <RHLogo />
                            <Text style={appliedStyles.currDate}>
                                Prepared {DateFormat({ date: new Date(), type: 'exact' }).props.children}
                            </Text>
                        </View>
                        <View>
                            <Text style={[ appliedStyles.reportName, appliedStyles.largeSpacing, appliedStyles.displayFont ]}>
                                {reportName} {type}
                            </Text>
                        </View>
                        <View>
                            <Text style={ appliedStyles.displayFont }>
                                { customTitle(title) }
                            </Text>

                        </View>
                    </React.Fragment>}
                    <View style={appliedStyles.smallSpacing}>
                        {page}
                    </View>
                    <View style={[ appliedStyles.flexRow, {
                        marginTop: 'auto'
                    }]}>
                        <Text style={[{
                            marginLeft: 'auto'
                        }]}>
                            { key + 1 } of {pages.length}
                        </Text>
                        <Text style={[{
                            marginLeft: 'auto'
                        }, appliedStyles.thirdTitle ]}>
                            redhat.com
                        </Text>
                    </View>
                </Page>
            ))}
        </Document>;

        return (
            <React.Fragment>
                { isPreview ? <PDFViewer>
                    {document}
                </PDFViewer> : <PDFDownloadLink
                    document={document}
                    fileName={fileName}
                    {...props}
                >
                    {label}
                </PDFDownloadLink> }
            </React.Fragment>
        );
    }
}

DownloadButton.propTypes = {
    pages: PropTypes.arrayOf(PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.arrayOf(PropTypes.node)
    ])),
    reportName: PropTypes.string,
    fileName: PropTypes.string,
    type: PropTypes.string,
    title: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.arrayOf(PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.shape({
                title: PropTypes.string,
                fontWeight: PropTypes.number,
                style: PropTypes.shape({
                    [PropTypes.string]: PropTypes.oneOfType([ PropTypes.number, PropTypes.string ])
                })
            })
        ]))
    ]),
    isPreview: PropTypes.bool,
    label: PropTypes.node
};

DownloadButton.defaultProps = {
    reportName: 'Executive report:',
    pages: [],
    title: '',
    type: 'Default',
    fileName: '',
    label: 'Download PDF',
    isPreview: false
};

export default DownloadButton;
