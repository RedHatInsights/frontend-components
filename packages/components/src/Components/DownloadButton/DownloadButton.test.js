import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DownloadButton from './DownloadButton';
import { DropdownItem } from '@patternfly/react-core';

const extraItems = [
    <DropdownItem key="extra-1" component="button"></DropdownItem>
];

describe('DownloadButton component', () => {
    describe('should render', () => {
        it('CSV and JSON by default', () => {
            const wrapper = shallow(<DownloadButton />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('custom items', () => {
            const wrapper = shallow(<DownloadButton extraItems={extraItems} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('API', () => {
        it('clicking should open dropdown', () => {
            const wrapper = mount(<DownloadButton extraItems={extraItems} />);
            wrapper.find('button').first().simulate('click');
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('onSelect should be called with CSV', () => {
            const onSelect = jest.fn();
            const wrapper = mount(<DownloadButton extraItems={extraItems} onSelect={onSelect} />);
            wrapper.find('button').first().simulate('click');
            wrapper.find('ul button').first().simulate('click');
            expect(onSelect.mock.calls.length).toBe(1);
            expect(onSelect.mock.calls[0][1]).toBe('csv');
        });

        it('onSelect should be called with JSON', () => {
            const onSelect = jest.fn();
            const wrapper = mount(<DownloadButton extraItems={extraItems} onSelect={onSelect} />);
            wrapper.find('button').first().simulate('click');
            wrapper.find('ul button').at(1).simulate('click');
            expect(onSelect.mock.calls.length).toBe(1);
            expect(onSelect.mock.calls[0][1]).toBe('json');
        });

        it('shouldn\'t call onSelect', () => {
            const onSelect = jest.fn();
            const wrapper = mount(<DownloadButton extraItems={extraItems} />);
            wrapper.find('button').first().simulate('click');
            wrapper.find('ul button').at(1).simulate('click');
            expect(onSelect.mock.calls.length).toBe(0);
        });
    });
});
