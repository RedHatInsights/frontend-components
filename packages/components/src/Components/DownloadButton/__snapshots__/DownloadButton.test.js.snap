// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadButton component API clicking should open dropdown 1`] = `
<DownloadButton
  extraItems={
    Array [
      <DropdownItem
        className=""
        component="button"
        href=""
        isDisabled={false}
        isHovered={false}
        tooltip={null}
        tooltipProps={Object {}}
      />,
    ]
  }
  onSelect={[Function]}
>
  <Dropdown
    className=""
    direction="down"
    dropdownItems={
      Array [
        <DropdownItem
          className=""
          component="button"
          href=""
          isDisabled={false}
          isHovered={false}
          onClick={[Function]}
          tooltip={null}
          tooltipProps={Object {}}
        >
          CSV
        </DropdownItem>,
        <DropdownItem
          className=""
          component="button"
          href=""
          isDisabled={false}
          isHovered={false}
          onClick={[Function]}
          tooltip={null}
          tooltipProps={Object {}}
        >
          JSON
        </DropdownItem>,
        <DropdownItem
          className=""
          component="button"
          href=""
          isDisabled={false}
          isHovered={false}
          tooltip={null}
          tooltipProps={Object {}}
        />,
      ]
    }
    isGrouped={false}
    isOpen={true}
    isPlain={true}
    onSelect={[Function]}
    position="left"
    toggle={
      <DropdownToggle
        className=""
        iconComponent={null}
        id=""
        isActive={false}
        isFocused={false}
        isHovered={false}
        isOpen={false}
        isPlain={false}
        onToggle={[Function]}
        parentRef={null}
      >
        <FileExportIcon
          color="currentColor"
          size="md"
          title={null}
        />
      </DropdownToggle>
    }
  >
    <DropdownWithContext
      className=""
      direction="down"
      dropdownItems={
        Array [
          <DropdownItem
            className=""
            component="button"
            href=""
            isDisabled={false}
            isHovered={false}
            onClick={[Function]}
            tooltip={null}
            tooltipProps={Object {}}
          >
            CSV
          </DropdownItem>,
          <DropdownItem
            className=""
            component="button"
            href=""
            isDisabled={false}
            isHovered={false}
            onClick={[Function]}
            tooltip={null}
            tooltipProps={Object {}}
          >
            JSON
          </DropdownItem>,
          <DropdownItem
            className=""
            component="button"
            href=""
            isDisabled={false}
            isHovered={false}
            tooltip={null}
            tooltipProps={Object {}}
          />,
        ]
      }
      isGrouped={false}
      isOpen={true}
      isPlain={true}
      onSelect={[Function]}
      position="left"
      toggle={
        <DropdownToggle
          className=""
          iconComponent={null}
          id=""
          isActive={false}
          isFocused={false}
          isHovered={false}
          isOpen={false}
          isPlain={false}
          onToggle={[Function]}
          parentRef={null}
        >
          <FileExportIcon
            color="currentColor"
            size="md"
            title={null}
          />
        </DropdownToggle>
      }
    >
      <div
        className="pf-c-dropdown pf-m-expanded"
      >
        <DropdownToggle
          ariaHasPopup={true}
          className=""
          iconComponent={null}
          id="pf-toggle-id-1"
          isActive={false}
          isFocused={false}
          isHovered={false}
          isOpen={true}
          isPlain={true}
          key=".0"
          onEnter={[Function]}
          onToggle={[Function]}
          parentRef={
            <div
              class="pf-c-dropdown pf-m-expanded"
            >
              <button
                aria-expanded="true"
                aria-haspopup="true"
                class="pf-c-dropdown__toggle pf-m-plain"
                id="pf-toggle-id-1"
                type="button"
              >
                <span>
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1.5em"
                    role="img"
                    style="vertical-align: -0.1875em;"
                    viewBox="0 0 576 512"
                    width="1.5em"
                  >
                    <path
                      d="M384 121.9c0-6.3-2.5-12.4-7-16.9L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128zM571 308l-95.7-96.4c-10.1-10.1-27.4-3-27.4 11.3V288h-64v64h64v65.2c0 14.3 17.3 21.4 27.4 11.3L571 332c6.6-6.6 6.6-17.4 0-24zm-379 28v-32c0-8.8 7.2-16 16-16h176V160H248c-13.2 0-24-10.8-24-24V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V352H208c-8.8 0-16-7.2-16-16z"
                      transform=""
                    />
                  </svg>
                </span>
              </button>
              <ul
                aria-labelledby="pf-toggle-id-1"
                class="pf-c-dropdown__menu"
                role="menu"
              >
                <li
                  role="none"
                >
                  <button
                    class="pf-c-dropdown__menu-item"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    CSV
                  </button>
                </li>
                <li
                  role="none"
                >
                  <button
                    class="pf-c-dropdown__menu-item"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    JSON
                  </button>
                </li>
                <li
                  role="none"
                >
                  <button
                    class="pf-c-dropdown__menu-item"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  />
                </li>
              </ul>
            </div>
          }
        >
          <Toggle
            ariaHasPopup={true}
            className=""
            id="pf-toggle-id-1"
            isActive={false}
            isDisabled={false}
            isFocused={false}
            isHovered={false}
            isOpen={true}
            isPlain={true}
            isSplitButton={false}
            onEnter={[Function]}
            onToggle={[Function]}
            parentRef={
              <div
                class="pf-c-dropdown pf-m-expanded"
              >
                <button
                  aria-expanded="true"
                  aria-haspopup="true"
                  class="pf-c-dropdown__toggle pf-m-plain"
                  id="pf-toggle-id-1"
                  type="button"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1.5em"
                      role="img"
                      style="vertical-align: -0.1875em;"
                      viewBox="0 0 576 512"
                      width="1.5em"
                    >
                      <path
                        d="M384 121.9c0-6.3-2.5-12.4-7-16.9L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128zM571 308l-95.7-96.4c-10.1-10.1-27.4-3-27.4 11.3V288h-64v64h64v65.2c0 14.3 17.3 21.4 27.4 11.3L571 332c6.6-6.6 6.6-17.4 0-24zm-379 28v-32c0-8.8 7.2-16 16-16h176V160H248c-13.2 0-24-10.8-24-24V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V352H208c-8.8 0-16-7.2-16-16z"
                        transform=""
                      />
                    </svg>
                  </span>
                </button>
                <ul
                  aria-labelledby="pf-toggle-id-1"
                  class="pf-c-dropdown__menu"
                  role="menu"
                >
                  <li
                    role="none"
                  >
                    <button
                      class="pf-c-dropdown__menu-item"
                      role="menuitem"
                      tabindex="-1"
                      type="button"
                    >
                      CSV
                    </button>
                  </li>
                  <li
                    role="none"
                  >
                    <button
                      class="pf-c-dropdown__menu-item"
                      role="menuitem"
                      tabindex="-1"
                      type="button"
                    >
                      JSON
                    </button>
                  </li>
                  <li
                    role="none"
                  >
                    <button
                      class="pf-c-dropdown__menu-item"
                      role="menuitem"
                      tabindex="-1"
                      type="button"
                    />
                  </li>
                </ul>
              </div>
            }
            type={null}
          >
            <button
              aria-expanded={true}
              aria-haspopup={true}
              className="pf-c-dropdown__toggle pf-m-plain"
              disabled={false}
              id="pf-toggle-id-1"
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              <span
                className={null}
              >
                <FileExportIcon
                  color="currentColor"
                  size="md"
                  title={null}
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1.5em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.1875em",
                      }
                    }
                    viewBox="0 0 576 512"
                    width="1.5em"
                  >
                    <path
                      d="M384 121.9c0-6.3-2.5-12.4-7-16.9L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128zM571 308l-95.7-96.4c-10.1-10.1-27.4-3-27.4 11.3V288h-64v64h64v65.2c0 14.3 17.3 21.4 27.4 11.3L571 332c6.6-6.6 6.6-17.4 0-24zm-379 28v-32c0-8.8 7.2-16 16-16h176V160H248c-13.2 0-24-10.8-24-24V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V352H208c-8.8 0-16-7.2-16-16z"
                      transform=""
                    />
                  </svg>
                </FileExportIcon>
              </span>
            </button>
          </Toggle>
        </DropdownToggle>
        <DropdownMenu
          aria-labelledby="pf-toggle-id-1"
          className=""
          component="ul"
          isGrouped={false}
          isOpen={true}
          openedOnEnter={false}
          position="left"
        >
          <ul
            aria-labelledby="pf-toggle-id-1"
            className="pf-c-dropdown__menu"
            hidden={false}
            role="menu"
          >
            <DropdownItem
              className=""
              component="button"
              href=""
              index={0}
              isDisabled={false}
              isHovered={false}
              key=".$download-csv"
              onClick={[Function]}
              tooltip={null}
              tooltipProps={Object {}}
            >
              <DropdownItem
                className=""
                component="button"
                context={
                  Object {
                    "keyHandler": [Function],
                    "sendRef": [Function],
                  }
                }
                href=""
                index={0}
                isDisabled={false}
                isHovered={false}
                onClick={[Function]}
                role="menuitem"
                tabIndex={-1}
                tooltip={null}
                tooltipProps={Object {}}
              >
                <li
                  role="none"
                >
                  <button
                    className="pf-c-dropdown__menu-item"
                    disabled={false}
                    href={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="menuitem"
                    tabIndex={-1}
                    type="button"
                  >
                    CSV
                  </button>
                </li>
              </DropdownItem>
            </DropdownItem>
            <DropdownItem
              className=""
              component="button"
              href=""
              index={1}
              isDisabled={false}
              isHovered={false}
              key=".$download-json"
              onClick={[Function]}
              tooltip={null}
              tooltipProps={Object {}}
            >
              <DropdownItem
                className=""
                component="button"
                context={
                  Object {
                    "keyHandler": [Function],
                    "sendRef": [Function],
                  }
                }
                href=""
                index={1}
                isDisabled={false}
                isHovered={false}
                onClick={[Function]}
                role="menuitem"
                tabIndex={-1}
                tooltip={null}
                tooltipProps={Object {}}
              >
                <li
                  role="none"
                >
                  <button
                    className="pf-c-dropdown__menu-item"
                    disabled={false}
                    href={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="menuitem"
                    tabIndex={-1}
                    type="button"
                  >
                    JSON
                  </button>
                </li>
              </DropdownItem>
            </DropdownItem>
            <DropdownItem
              className=""
              component="button"
              href=""
              index={2}
              isDisabled={false}
              isHovered={false}
              key=".$extra-1"
              tooltip={null}
              tooltipProps={Object {}}
            >
              <DropdownItem
                className=""
                component="button"
                context={
                  Object {
                    "keyHandler": [Function],
                    "sendRef": [Function],
                  }
                }
                href=""
                index={2}
                isDisabled={false}
                isHovered={false}
                onClick={[Function]}
                role="menuitem"
                tabIndex={-1}
                tooltip={null}
                tooltipProps={Object {}}
              >
                <li
                  role="none"
                >
                  <button
                    className="pf-c-dropdown__menu-item"
                    disabled={false}
                    href={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="menuitem"
                    tabIndex={-1}
                    type="button"
                  />
                </li>
              </DropdownItem>
            </DropdownItem>
          </ul>
        </DropdownMenu>
      </div>
    </DropdownWithContext>
  </Dropdown>
</DownloadButton>
`;

exports[`DownloadButton component should render CSV and JSON by default 1`] = `
<Dropdown
  className=""
  direction="down"
  dropdownItems={
    Array [
      <DropdownItem
        className=""
        component="button"
        href=""
        isDisabled={false}
        isHovered={false}
        onClick={[Function]}
        tooltip={null}
        tooltipProps={Object {}}
      >
        CSV
      </DropdownItem>,
      <DropdownItem
        className=""
        component="button"
        href=""
        isDisabled={false}
        isHovered={false}
        onClick={[Function]}
        tooltip={null}
        tooltipProps={Object {}}
      >
        JSON
      </DropdownItem>,
    ]
  }
  isGrouped={false}
  isOpen={false}
  isPlain={true}
  onSelect={[Function]}
  position="left"
  toggle={
    <DropdownToggle
      className=""
      iconComponent={null}
      id=""
      isActive={false}
      isFocused={false}
      isHovered={false}
      isOpen={false}
      isPlain={false}
      onToggle={[Function]}
      parentRef={null}
    >
      <FileExportIcon
        color="currentColor"
        size="md"
        title={null}
      />
    </DropdownToggle>
  }
/>
`;

exports[`DownloadButton component should render custom items 1`] = `
<Dropdown
  className=""
  direction="down"
  dropdownItems={
    Array [
      <DropdownItem
        className=""
        component="button"
        href=""
        isDisabled={false}
        isHovered={false}
        onClick={[Function]}
        tooltip={null}
        tooltipProps={Object {}}
      >
        CSV
      </DropdownItem>,
      <DropdownItem
        className=""
        component="button"
        href=""
        isDisabled={false}
        isHovered={false}
        onClick={[Function]}
        tooltip={null}
        tooltipProps={Object {}}
      >
        JSON
      </DropdownItem>,
      <DropdownItem
        className=""
        component="button"
        href=""
        isDisabled={false}
        isHovered={false}
        tooltip={null}
        tooltipProps={Object {}}
      />,
    ]
  }
  isGrouped={false}
  isOpen={false}
  isPlain={true}
  onSelect={[Function]}
  position="left"
  toggle={
    <DropdownToggle
      className=""
      iconComponent={null}
      id=""
      isActive={false}
      isFocused={false}
      isHovered={false}
      isOpen={false}
      isPlain={false}
      onToggle={[Function]}
      parentRef={null}
    >
      <FileExportIcon
        color="currentColor"
        size="md"
        title={null}
      />
    </DropdownToggle>
  }
/>
`;
